src/main/java/com/google/devtools/build/lib/bazel/bzlmod/InnateRunnableExtension.java

```diff
@@ -96,62 +93,56 @@ static InnateRunnableExtension load(
-        throw ExternalDepsException.withCauseAndMessage(
-            Code.BAD_MODULE, e, "bad repo rule .bzl file label at %s", tag.getLocation());
-      }
+
+    // The name of the extension is of the form "<bzl_file_label>%<rule_name>".
+    Iterator<String> parts = Splitter.on('%').split(extensionId.getExtensionName()).iterator();
+    // ModuleFileFunction doesn't add usages for use_repo_rule without any instantiations, so we can
+    // assume that there is at least one tag.
+    Location location = tags.getFirst().getLocation();
+    Label bzlLabel;
+    try {
+      bzlLabel = Label.parseWithRepoContext(parts.next(), repoContext);
+      BzlLoadFunction.checkValidLoadLabel(bzlLabel, starlarkSemantics);
+    } catch (LabelSyntaxException e) {
+      throw ExternalDepsException.withCauseAndMessage(
+          Code.BAD_MODULE, e, "bad repo rule .bzl file label at %s", location);
     }
-    ImmutableSet<BzlLoadValue.Key> loadKeys =
-        repoBuilders.stream()
-            .map(r -> BzlLoadValue.keyForBzlmod(r.bzlLabel()))
-            .collect(toImmutableSet());
-    HashSet<Label> digestedLabels = new HashSet<>();
-    Fingerprint transitiveBzlDigest = new Fingerprint();
-    SkyframeLookupResult loadResult = env.getValuesAndExceptions(loadKeys);
-    for (InnateExtensionRepo.Builder repoBuilder : repoBuilders) {
-      BzlLoadValue loadedBzl;
-      try {
-        loadedBzl =
-            (BzlLoadValue)
-                loadResult.getOrThrow(
-                    BzlLoadValue.keyForBzlmod(repoBuilder.bzlLabel()),
-                    BzlLoadFailedException.class);
-      } catch (BzlLoadFailedException e) {
-        throw ExternalDepsException.withCauseAndMessage(
-            Code.BAD_MODULE,
-            e,
-            "error loading '%s' for repo rules, requested by %s",
-            repoBuilder.bzlLabel(),
-            repoBuilder.tag().getLocation());
-      }
-      if (loadedBzl == null) {
-        return null;
-      }
-      repoBuilder.loadedBzl(loadedBzl);
-      if (digestedLabels.add(repoBuilder.bzlLabel())) {
-        // Only digest this BzlLoadValue if we haven't seen this bzl label before.
-        transitiveBzlDigest.addBytes(loadedBzl.getTransitiveDigest());
-      }
+    String ruleName = parts.next();
+
+    // Load the .bzl file.
+    BzlLoadValue loadedBzl;
+    try {
+      loadedBzl =
+          (BzlLoadValue)
+              env.getValueOrThrow(
+                  BzlLoadValue.keyForBzlmod(bzlLabel), BzlLoadFailedException.class);
+    } catch (BzlLoadFailedException e) {
+      throw ExternalDepsException.withCauseAndMessage(
+          Code.BAD_MODULE,
+          e,
+          "error loading '%s' for repo rules, requested by %s",
+          bzlLabel,
+          location);
     }
+    if (loadedBzl == null) {
+      return null;
+    }
+
+    ImmutableList<InnateExtensionRepo> repos =
+        tags.stream()
+            .map(
+                tag ->
+                    InnateExtensionRepo.builder()
```

Comment:
it doesn't look like this builder needs to exist anymore. Actually this record doesn't really need to exist anymore -- the `bzlLabel`, `ruleName`, and `loadedBzl` are all shared by all the repos, so only the `tag` is different. we can just store a list of `tag`s and put the common data into `InnateRunnableExtension` itself.